### PR TITLE
:tada: New Source: Mailjet SMS

### PR DESCRIPTION
**Pull request by haithem-souala**
## What
https://dev.mailjet.com/sms

## How
New source connector for Mailjet SMS API

## Streams
1. SMS

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

Capture d’écran 2022-10-24 à 00 02 57

</details>

### CODE REVIEW

**Comment by haithem-souala** [comment]
Capture d’écran 2022-10-24 à 00 02 57

**Comment by marcosmarxm** [comment]
Thanks for the contribution! Someone will review it during this week. If you have any questions please create comments so the reviewer can focus on those points.

File: airbyte-integrations/connectors/source-mailjet-sms/integration_tests/catalog.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  "streams": [
+    {
+      "name": "TODO fix this file",
+      "supported_sync_modes": ["full_refresh", "incremental"],
+      "source_defined_cursor": true,
+      "default_cursor_field": "column1",
+      "json_schema": {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "type": "object",
+        "properties": {
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-sms/integration_tests/catalog.json:11]: Remove this file.

File: airbyte-integrations/connectors/source-mailjet-sms/source_mailjet_sms/mailjet_sms.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    authenticator:
+      type: BearerAuthenticator
+      api_token: "{{ config['token'] }}"
+  offset_paginator:
+    type: DefaultPaginator
+    $options:
+      url_base: "*ref(definitions.requester.url_base)"
+    pagination_strategy:
+      type: "OffsetIncrement"
+      page_size: 10
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-sms/source_mailjet_sms/mailjet_sms.yaml:19]: The limit is 100. Please use always the limit when possible. 
https://dev.mailjet.com/sms/guides/filtering-resources/#the-limit-filter

File: airbyte-integrations/connectors/source-mailjet-sms/source_mailjet_sms/mailjet_sms.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    page_token_option:
+      field_name: "offset"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-sms/source_mailjet_sms/mailjet_sms.yaml:21]: The name is correct? Is it not `Offset`?

File: airbyte-integrations/connectors/source-mailjet-sms/source_mailjet_sms/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  required:
+    - token
+  additionalProperties: true
+  properties:
+    token:
+      type: string
+      description: >- 
+        Your access token. See <a
+        href="https://dev.mailjet.com/sms/reference/overview/authentication">here</a>.
+      airbyte_secret: true
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-sms/source_mailjet_sms/spec.yaml:15]: The API allows you to use start date and end date, are you planning to add those parameters?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments! But looks great.

**Comment by haithem-souala** [comment]
Capture d’écran 2022-10-25 à 14 47 57

File: airbyte-integrations/connectors/source-mailjet-sms/integration_tests/sample_config.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+  "fix-me": "TODO"
+}
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-sms/integration_tests/sample_config.json:3]: Please fix this file.

File: airbyte-integrations/connectors/source-mailjet-sms/source_mailjet_sms/spec.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-sms/source_mailjet_sms/spec.yaml:15]:
```suggestion
    token:
      title: Access Token
      type: string
      description: >- 
        Your access token. See <a
        href="https://dev.mailjet.com/sms/reference/overview/authentication">here</a>.
      airbyte_secret: true
```
Please add title to all parameters.

File: airbyte-integrations/connectors/source-mailjet-sms/source_mailjet_sms/spec.yaml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+documentationUrl: https://dev.mailjet.com/sms
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-sms/source_mailjet_sms/spec.yaml:1]: update this to direct to Airbyte Source connector url

File: airbyte-integrations/connectors/source-mailjet-sms/source_mailjet_sms/spec.yaml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+      type: string
+      description: >- 
+        Your access token. See <a
+        href="https://dev.mailjet.com/sms/reference/overview/authentication">here</a>.
+      airbyte_secret: true
+    start_date:
+      type: integer
+      description: Retrieve SMS messages created after the specified timestamp. Required format - Unix timestamp.
+      examples:
+        - 1666261656
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-mailjet-sms/source_mailjet_sms/spec.yaml:20]: Can you add regex pattern to make sure people add the correct input?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Almost ready! A few comments. Ask my review again after comments are solved.

**Review by marcosmarxm — APPROVED** [review]
I'll publish this tomorrow! Congrats @haithem-souala !

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**